### PR TITLE
[AIRFLOW-4430] Fix "Zoom into Sub DAG" link

### PR DESCRIPTION
**Pull request by marcusianlevine**
### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-4430

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Fixes the "Zoom into Sub DAG" button to link to the proper URL, and fixes an associated error when looking up the DAG object in the `tree` view for a SubDAG

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Added a unit test for the `tree` view for a subDAG

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -1149,11 +1149,10 @@ def tree(self):
         default_dag_run = conf.getint('webserver', 'default_dag_run_display_number')
         dag_id = request.args.get('dag_id')
         blur = conf.getboolean('webserver', 'demo_mode')
-        dag_model = DagModel.get_dagmodel(dag_id)
-        if not dag_model:
-            flash('DAG "{0}" seems to be missing in database.'.format(dag_id), "error")
+        dag = dagbag.get_dag(dag_id)
```
Comment (ashb) [inline airflow/www/views.py]: We'd like to stay using the DagModel if possible. Is this change required?

Comment (marcusianlevine) [reply airflow/www/views.py]: For the Graph view, no, but the tree view seems to be broken for SubDAGs

The issue seems to arise from the `dag` object being `None` when using `DagModel.get_dagmodel` for a `dag_id` that corresponds to a subDAG

We could merge this PR without the change from `DagModel`, but then if people are using `tree` as their default view the "Zoom into Sub DAG" link will result in an error

I switched to `dagbag.get_dag` because that's the method used by the `graph` view and that view works with subDAGs

Comment (ashb) [reply airflow/www/views.py]: Ah right, sounds sensible.

Comment (marcusianlevine) [reply airflow/www/views.py]: I switched it to "fall back" to the `DagBag` approach instead of replacing `DagModel` wholesale, let me know if that seems appropriate

**Review by kaxil — APPROVED** [review]

File: tests/www/test_views.py [hunk 2]
```diff
@@ -474,6 +474,11 @@ def test_tree(self):
         resp = self.client.get(url, follow_redirects=True)
         self.check_content_in_response('runme_1', resp)
 
+    def test_tree_subdag(self):
+        url = 'tree?dag_id=example_subdag_operator.section-1'
+        resp = self.client.get(url, follow_redirects=True)
+        self.check_content_in_response('section-1', resp)
```
Comment (ashb) [inline tests/www/test_views.py]: This message would appear even even if the dag couldn't be found (from the "Dag X couldn't be found" message) - could we check for something else in the content?

Comment (marcusianlevine) [reply tests/www/test_views.py]: Sure, switched to the name of one of the tasks inside the subDAG

**Comment by marcusianlevine** [comment]
@ashb I made the requested changes to the test, think this is ready for merge?

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
Thanks @marcusianlevine!

**Comment by ashb** [comment]
@marcusianlevine I'm not able to reproduce this on 1.10.3 - Can you confirm if this is a problem there or only against master?

Classic UI code: https://github.com/apache/airflow/blob/16d93c9e45e14179c7822fed248743f0c3fd935c/airflow/www/templates/airflow/dag.html#L423-L428

RBAC UI code: https://github.com/apache/airflow/blob/16d93c9e45e14179c7822fed248743f0c3fd935c/airflow/www_rbac/templates/airflow/dag.html#L425-L430

**Comment by marcusianlevine** [comment]
You're correct, this issue is not present in 1.10.3 but was introduced on master after 1.10.3 was released

I initially discovered this bug while working on #5185
